heat/core/dndarray.py

```diff
@@ -778,16 +823,24 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
                 if not isinstance(lkey[self.split], torch.Tensor)
                 else lkey[self.split]
             )
-
             loc_inds = torch.where((inds >= chunk_start) & (inds < chunk_end))
+            # if there are no local indices on a process, then `arr` is empty
```

Comment:
> if there are no local indices on a process, then `arr` is empty

`arr` is empty in the `split` dimension -> `lout[self.split]` should be set to 0 here

Reply:
You're right, done.